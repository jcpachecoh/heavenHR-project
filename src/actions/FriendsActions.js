import * as types from '../constants/ActionTypes';

export function addFriend(name, sex) {
  console.log("dentro funcion"+sex);
  return {
    type: types.ADD_FRIEND,
    name, 
    sex
  };
}

export function deleteFriend(id) {
  console.log(id);
  return {
    type: types.DELETE_FRIEND,
    id
  };
}

export function starFriend(id) {
  return {
    type: types.STAR_FRIEND,
    id
  };
}
