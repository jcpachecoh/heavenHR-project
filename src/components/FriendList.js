import React, { Component, PropTypes } from 'react';
import styles from './FriendList.css';
import FriendListItem from './FriendListItem';

class FriendList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 1,
      friendsPerPage: 2,
      friends: this.props.friends
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(event) {
    this.setState({
      currentPage: Number(event.target.id)
    });
  }

  render () {
    const { currentPage, friendsPerPage } = this.state;

    const indexOfLastFriend = currentPage * friendsPerPage;
    const indexOfFirstFriend= indexOfLastFriend - friendsPerPage;
    const currentFriends = this.props.friends.slice(indexOfFirstFriend, indexOfLastFriend);

    const renderFriends = currentFriends.map((friend, index) => {
      let indexFriend;
      if(currentPage !== 1) 
        indexFriend = index + currentPage;
      else
        indexFriend = index;

      return (
        <FriendListItem
          key={indexFriend}
          id={indexFriend}
          name={friend.name}
          sex={friend.sex}
          starred={friend.starred}
          {...this.props.actions} />
      );
    });

    const pageNumbers = [];
    for (let i = 1; i <= Math.ceil(this.props.friends.length / friendsPerPage); i++) {
      pageNumbers.push(i);
    }

    const renderPageNumbers = pageNumbers.map(number => {
      return (
        <li
          key={number}
          id={number}
          onClick={this.handleClick}>
          {number}
        </li>
      );
    });

    return (<div>
        <ul className={styles.friendList}>
          {renderFriends}
        </ul>
        <ul id="page-numbers">
        {renderPageNumbers}
      </ul>
    </div>
    );
  }

}

FriendList.propTypes = {
  friends: PropTypes.array.isRequired,
  actions: PropTypes.object.isRequired
};

export default FriendList;
