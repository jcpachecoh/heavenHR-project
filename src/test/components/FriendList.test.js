import React from 'react';
import FriendList from '../../components/FriendList';
import * as TestUtils from 'react-addons-test-utils';
import {addFriend, deleteFriend, starFriend} from '../../actions/FriendsActions';

describe('Testing <FriendList /> component', () => {
    it('Should render FriendList component ', () => {
        const friendsMockData = {
            friendsById: [
              {
                name: 'Theodore Roosevelt',
                sex: 'Male',
                starred: true
              },
              {
                name: 'Abraham Lincoln',
                sex: 'Male',
                starred: false
              },
              {
                name: 'George Washington',
                sex: 'Female',
                starred: false
              }
            ]
          };

        const actions = {
            addFriend: addFriend,
            deleteFriend: deleteFriend,
            starFriend: starFriend
        }; 
        expect(TestUtils.createRenderer().render(
            <FriendList friends = {friendsMockData.friendsById} actions = {actions}  />
          )).toMatchSnapshot()
    });
});
