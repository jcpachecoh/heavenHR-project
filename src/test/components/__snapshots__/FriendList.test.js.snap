// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FriendList /> component Should render FriendList component  1`] = `
<div>
  <ul
    className={undefined}
  >
    <FriendListItem
      addFriend={[Function]}
      deleteFriend={[Function]}
      id={0}
      name="Theodore Roosevelt"
      sex="Male"
      starFriend={[Function]}
      starred={true}
    />
    <FriendListItem
      addFriend={[Function]}
      deleteFriend={[Function]}
      id={1}
      name="Abraham Lincoln"
      sex="Male"
      starFriend={[Function]}
      starred={false}
    />
  </ul>
  <ul
    id="page-numbers"
  >
    <li
      id={1}
      onClick={[Function]}
    >
      1
    </li>
    <li
      id={2}
      onClick={[Function]}
    >
      2
    </li>
  </ul>
</div>
`;
